Makefile

```diff
@@ -21,46 +21,68 @@
 
 ###################################################################################
 
-# See https://tech.davis-hansson.com/p/make/ for more information
-# about these initial lines.
+# Some modules build their dependencies from variables, we want these to be 
+# evalutated at the last possible moment. For this we use second expansion to 
+# re-evaluate the generate and verify targets a second time.
+#
+# See https://www.gnu.org/software/make/manual/html_node/Secondary-Expansion.html
+.SECONDEXPANSION:
+
+# For details on some of these "prelude" settings, see:
+# https://clarkgrubb.com/makefile-style-guide
 MAKEFLAGS += --warn-undefined-variables --no-builtin-rules
 SHELL := /usr/bin/env bash
-.SHELLFLAGS := -eu -o pipefail -c
+.SHELLFLAGS := -uo pipefail -c
```

Comment:
note: this is changed to match https://github.com/cert-manager/makefile-modules/blob/140169ff41d87878ddb0fdfd5ecf567aee25d992/modules/repository-base/base/Makefile which is used in other repos. It's not really required and not really related to the rest of the PR but it's done for conformity.